src/manual/design/apis/pulse.md

```diff
@@ -10,20 +10,97 @@ communicate between microservices, notifying interested parties when specific
 events have occurred. The system is based on AMQP in a publish-subscribe
 configuration.
 
-All Taskcluster AMQP exchanges are `topic` exchanges. Message bodies are always
-JSON, and messages are "copied" to multiple routing keys using the AMQP `Cc`
-header.
+Pulse is a publicly accessible service, and the pulse message schemas are a
+part of Taskcluster's published API, so they make a nicely decoupled
+integration point for external services.
+
+## Background
+
+Mozilla, where Taskcluster began, has an organization-wide "message bus" called
+"pulse". The idea is that systems within Mozilla should publish events to this
+bus, and other systems can then consume those events. This has enabled
+development of lots of interesting pieces of automation. It was a natural
+choice to publish Taskcluster-related events to this message bus, and to also
+consume such events from the bus.
+
+That said, Taskcluster does not consume events from other Mozilla services, and
+can use any other RabbitMQ service just as easily.  It does use some
+RabbitMQ-specific extensions. It follows the [pulse
+specification](https://wiki.mozilla.org/Auto-tools/Projects/Pulse#Specification),
+so any RabbitMQ service configured in a manner compatible with that specification
+should work.
+
+Consider the use of the term "pulse" within Taskcluster to mean
+"pulse-compatible RabbitMQ".
+
+### Pulse Specficiation, Abbreviated
+
+The pulse specification is summarized here for posterity:
+
+---
+
+Pulse (the server):
+
+* MUST support AMQP 0-9-1 and these RabbitMQ extensions:
+
+ * Confirms
+ * Consumer Prefetch
+ * Queue Length Limit
+ * Sender-selected Distribution
+
+* SHOULD exhibit deliver-at-least-once semantics
+* MAY delete queues that grows beyond Pulse defined limits
+* SHALL notify owner by email when a queue grows close to Pulse-defined limits.
+
+Publishers:
+
+* SHOULD use confirm-publish channels
 
-Usually, we prefix all exchanges from the same component with a common
-`exchangePrefix` to ensure uniqueness. For routing keys we strive to always
-have the same entries for messages on a given exchange and use `_` if no
-value makes sense for the given routing key entry with a specific message.
+Exchanges:
+
+* MUST be named exchange/<clientId>/<name>
```

Comment:
Probably use back-ticks... and a different name that `clientId`, maybe `namespace` or `username`

Reply:
I just copied this from the wiki.. I'm not sure editing is useful?

Reply:
Ship it... We'll revise when we do tc-pulse...

As something here might need additional explaining...